packages/forge/blocks/openai/actions/createChatCompletion.tsx

```diff
@@ -194,33 +197,44 @@ export const createChatCompletion = createAction({
 
       const messages = parseChatCompletionMessages({ options, variables })
 
-      const response = await openai.chat.completions.create({
+      const body = {
         model: options.model ?? defaultOpenAIOptions.model,
         temperature: options.temperature
           ? Number(options.temperature)
           : undefined,
         messages,
         tools,
-      })
+      }
 
-      let message = response.choices[0].message
-      let totalTokens = response.usage?.total_tokens
+      let totalTokens = 0
+      let message: ChatCompletionMessage
+      while (true) {
+        const response = await openai.chat.completions.create(body)
+
+        message = response.choices[0].message
+        totalTokens += response.usage?.total_tokens || 0
+
+        if (!message.tool_calls) {
+          break
+        }
 
-      if (message.tool_calls) {
         messages.push(message)
 
         for (const toolCall of message.tool_calls) {
           const name = toolCall.function?.name
           if (!name) continue
           const toolDefinition = options.tools?.find((t) => t.name === name)
           if (!toolDefinition?.code || !toolDefinition.parameters) continue
+          const toolParams = Object.fromEntries(
+            toolDefinition.parameters.map(({ name }) => [name, null])
+          )
```

Comment:
Can you explain why you are adding this?

Reply:
If the LLM decides not to use an optional parameter, we'll encounter an error because the variable will not be defined. The objective of this code is to ensure that all variables are defined (and set to null by default).